### ISSUE TITLE
AWS Systems Manager Parameter Store backend support

### ISSUE THREAD

**Issue by melnikovn** [argoproj-labs/argocd-vault-plugin#466]
**Is your feature request related to a problem? Please describe.**
AWS Systems Manager Parameter Store can be used as a cheaper alternative to Secret Manager. It would be nice if AVP had support for it.

**Describe the solution you'd like**
I'd like to grab parameter values (that can be actually encrypted) in a way similar to other backends

https://docs.aws.amazon.com/systems-manager/latest/userguide/systems-manager-parameter-store.html

**Comment by werne2j** [comment]
@melnikovn Thanks for the suggestion, we might not be able to get to this but feel free to put together a PR and we would be happy to review it

**Comment by oujonny** [comment]
I started working on it. But my coding skills are limited, so any feedback is appreciated. I will open an PR soon.

https://github.com/oujonny/argocd-vault-plugin/tree/AWSSSMParameterStoreBackend

**Comment by greyhats13** [comment]
This feature is really awesome, I'm sure the community will love it. Is it ready to use? 

> I started working on it. But my coding skills are limited, so any feedback is appreciated. I will open an PR soon. 😄
> 
> https://github.com/oujonny/argocd-vault-plugin/tree/AWSSSMParameterStoreBackend

**Comment by oujonny** [comment]
Yes, from my side it's ready. but I don't use it in production yet. But feel free to use it. I don't know why my PR is not getting looked at :(

**Comment by strophy** [comment]
I'd love to see this super useful feature implemented as well, but unfortunately it doesn't seem your PR will be merged anytime soon :disappointed: I wonder if an alternative for now could be to use [external-secrets](https://github.com/external-secrets/external-secrets), which [supported by ArgoCD](https://argo-cd.readthedocs.io/en/stable/operator-manual/secret-management/) and in turn provides support for [AWS SSM Parameter Store](https://external-secrets.io/latest/introduction/stability-support/)?

I'm just beginning to learn about Kubernetes and ArgoCD, so I'd appreciate feedback on this approach, and will post back here if I try it myself and manage to get it working.

**Comment by MrSaints** [comment]
@strophy I've been trialling External Secrets, and it works great. I'm not sure what value ArgoCD's Vault Plugin gives over that, but from what I can see, External Secrets should probably meet your requirements.

**Comment by strophy** [comment]
Thanks @MrSaints I gave external-secrets a try and it was super simple and worked first try, can highly recommend it for anyone needing to get data from Parameter Store.
